std/range/package.d

```diff
@@ -6562,17 +6561,23 @@ if (isForwardRange!RangeOfRanges &&
             if (!e.empty)
             {
                 e.popFront();
-                r.front = e;
+                static if (hasAssignableElements!(ElementType!RangeOfRanges))
+                {
+                    r.front = e;
+                }
             }
 
             r.popFront();
         }
     }
 
-    // ElementType opIndex(size_t n)
-    // {
-    //     return _input[n].front;
-    // }
+    static if (isRandomAccessRange!RangeOfRanges && isRandomAccessRange!(ElementType!RangeOfRanges))
+    {
+        auto opIndex(size_t n)
+        {
+            return TransposedIndex!RangeOfRanges(_input, n);
```

Comment:
Maybe this is a stupid question, but why not just return the range at the `n`th position? That way you can still have `opIndex` even if the `ElementType!RangeOfRanges` isn't random access.